### PR TITLE
itR-17: add weather query by city name

### PR DESCRIPTION
**Pull request by MilaKolasava**
image
image

### CODE REVIEW

File: src/components/CityWeather/CityWeather.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  return (
+    <div className="cityweather-wrapper">
+      <div className="cityweather-search">
+        <input
+          value={city}
+          onChange={(event) => setCity(event.target.value)}
+          placeholder="Enter City"
+          type="text"
+        />
+        <Loupa
```
Comment (glebpigulevsky) [inline src/components/CityWeather/CityWeather.tsx]: За...

Comment (glebpigulevsky) [reply src/components/CityWeather/CityWeather.tsx]: Loupe !!!

File: src/components/CityWeather/CityWeather.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      </div>
+      {isLoading ? (
+        <LoadingSpinner />
+      ) : (
+        data && (
+          <div className="cityweather-container">
+            <div className="cityweather-location">
+              <b>{data && data.name}</b>
+            </div>
+            <div className="cityweather-tempepature">
```
Comment (glebpigulevsky) [inline src/components/CityWeather/CityWeather.tsx]: misspelling

File: src/components/CityWeather/CityWeather.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          setCity("");
+          setIsLoading(false);
+        })
+        .catch((error) => {
+          setError(error.message);
+        }),
+        [];
+    });
+  };
+  return (
```
Comment (glebpigulevsky) [inline src/components/CityWeather/CityWeather.tsx:14]: empty line before return

File: src/components/CityWeather/CityWeather.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    setIsLoading(true);
+
+    useEffect(() => {
+      axios
+        .get<CityWeatherResponse>(
+          `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${API_KEY}&units=metric`
+        )
+        .then((response) => {
+          setData(response.data);
+          console.log(response.data);
```
Comment (glebpigulevsky) [inline src/components/CityWeather/CityWeather.tsx]: no console.log in production!

**Comment by glebpigulevsky** [comment]
tests!

File: src/components/CityWeather/CityWeather.tsx (hunk 1 above)
Comment (MilaKolasava) [reply src/components/CityWeather/CityWeather.tsx]: Could this be written?

File: src/components/CityWeather/CityWeather.tsx (hunk 2 above)
Comment (MilaKolasava) [reply src/components/CityWeather/CityWeather.tsx]: fixed

File: src/components/CityWeather/CityWeather.tsx (hunk 3 above)
Comment (MilaKolasava) [reply src/components/CityWeather/CityWeather.tsx:14]: fix

File: src/components/CityWeather/CityWeather.tsx (hunk 4 above)
Comment (MilaKolasava) [reply src/components/CityWeather/CityWeather.tsx]: ok

File: src/components/CityWeather/CityWeather.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      });
+  };
+
+  useEffect(() => {
+    if (!!city) {
+      setIsEmpty(false);
+    }
+  }, [city]);
+
+  const getErrorView = () => {
```
Comment (glebpigulevsky) [inline src/components/CityWeather/CityWeather.tsx]: separate component please. its easy to test and easy to read

File: src/components/CityWeather/CityWeather.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            !isEmpty ? "enabled" : "disabled"
+          }`}
+          onClick={handleClick}
+        />
+      </div>
+
+      {isLoading ? (
+        <LoadingSpinner />
+      ) : error ? (
+        getErrorView()
```
Comment (glebpigulevsky) [inline src/components/CityWeather/CityWeather.tsx]: <ErrorView />

File: src/components/Weather/Weather.tsx [hunk 7]
```diff
@@ -58,7 +58,7 @@ function Weather() {
           <div className="weather-details" data-testid="weather-details">
             <p>
               <b>{intl.formatMessage({ id: "temperature" })}: </b>
-              {data.main.temp}°C
+              {data.main.temp.toFixed()}°C
```
Comment (glebpigulevsky) [inline src/components/Weather/Weather.tsx:61]: what does this method do? i have never seen it before

Comment (MilaKolasava) [reply src/components/Weather/Weather.tsx:61]: rounds the value to an integer

File: src/utils/getErrorView.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React, { useState } from "react";
+
+type CityWeatherError = {
+  response: any;
+  message: string;
+};
+
+function getErrorView() {
```
Comment (glebpigulevsky) [inline src/utils/getErrorView.tsx]: why lowecase name? why get?

File: src/utils/getErrorView.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  const [error, setError] = useState<CityWeatherError | null>(null);
```
Comment (glebpigulevsky) [inline src/utils/getErrorView.tsx]: why do you need state here?

File: src/utils/getErrorView.tsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React, { useState } from "react";
```
Comment (glebpigulevsky) [inline src/utils/getErrorView.tsx]: why do you move it in the utils folder? id doesnt look as utility, its fullf-fledged component

File: src/utils/getErrorView.tsx [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+  message: string;
+};
+
+function getErrorView() {
+  const [error, setError] = useState<CityWeatherError | null>(null);
+
+  return (
+    <div className="cityweather-errorView">
+      <p>
+        <b>Oh no, {error != null ? error.response.data.message : ""}.</b>
```
Comment (glebpigulevsky) [inline src/utils/getErrorView.tsx]: what is about translations?

**Review by glebpigulevsky — APPROVED** [review]
good job

File: src/components/CityWeather/CityWeather.tsx (hunk 5 above)
Comment (MilaKolasava) [reply src/components/CityWeather/CityWeather.tsx]: done

File: src/utils/getErrorView.tsx (hunk 8 above)
Comment (MilaKolasava) [reply src/utils/getErrorView.tsx]: fixed

File: src/utils/getErrorView.tsx (hunk 9 above)
Comment (MilaKolasava) [reply src/utils/getErrorView.tsx]: no

File: src/utils/getErrorView.tsx (hunk 10 above)
Comment (MilaKolasava) [reply src/utils/getErrorView.tsx]: it was the my mistake

File: src/utils/getErrorView.tsx (hunk 11 above)
Comment (MilaKolasava) [reply src/utils/getErrorView.tsx]: done

**Pull request merged**